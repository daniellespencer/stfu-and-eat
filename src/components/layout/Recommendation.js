import React, { Fragment, useState, useEffect } from 'react';
import { Jumbotron, Container } from 'react-bootstrap';
import RecommendationItem from './RecommendationItem';

const Recommendation = (props) => {

    const [restaurant, updateResults] = useState([]);
    // eslint-disable-next-line
    
    useEffect(() => {
        fetch('/api/restaurant-recommendation', {
            headers: {
                "Accept": 'application/json'
            }
        })
        .then(response => response.json().then(data => {
            updateResults(data.result)
            
            
            })
            
        );
    }, []);

    return (
        <Fragment>
        <Jumbotron fluid >
            <Container >
                <h1>Our Recommendation</h1>
                <p>
                We did it. We made a decision for you.
                </p>
            </Container>
        </Jumbotron>
        <RecommendationItem restaurant={restaurant}  />
        </Fragment>
    )
}

export default Recommendation
