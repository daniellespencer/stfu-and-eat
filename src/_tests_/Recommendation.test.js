import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';

import Recommendation from '../components/layout/Recommendation';

afterEach(cleanup);

it("renders correctly", () => {
    shallow(<Recommendation />);
})