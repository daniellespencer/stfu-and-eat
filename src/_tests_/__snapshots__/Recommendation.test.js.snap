// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recommendation renders restaurant recommendation 1`] = `
Array [
  <div
    className="jumbotron jumbotron-fluid"
  >
    <div
      className="container"
    >
      <h1>
        Our Recommendation
      </h1>
      <p>
        We did it. We made a decision for you.
      </p>
    </div>
  </div>,
  <div>
    <div
      className="shadow mb-5 mt-10  card"
    >
      <img
        className="card-img-top img-fluid"
        src="logo.jpg"
      />
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        />
        <p
          className="card-text"
        >
          Neighborhood: 
           
          <br />
          Cuisine: 
        </p>
        <button
          className="btn-primary btn-sm rec-item-button btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Restaurant website
        </button>
        <button
          className="btn-sm rec-item-button-directions btn btn-outline-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Directions
        </button>
      </div>
    </div>
  </div>,
]
`;
